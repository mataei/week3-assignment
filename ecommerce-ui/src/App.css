h1 {
  font-size: 1em;
  border-bottom: 1px solid #c9cac9;
}

h1 span {
  border-bottom: 1px solid #1d1b1b;
}

a { 
  text-decoration: none; 
  color: #1d1b1b;
}

img { 
  width: 100%; 
  height: auto; 
}

.allVacationRentals {
  display: grid;
  grid-template-rows: auto;
  grid-column-gap: 1em;
  grid-row-gap: 1em;
}

#allRentals {
  margin: 30px;
}

#cart {
  grid-template-columns: 1fr 1fr;
}

#cart .vacationTitle, #cart .vacationHouseType, #cart .vacationLocation {
  grid-column: 2;
}

#cart .article a {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-template-rows: auto;
  padding-right: 30px;
}

#cart .vacationImage {
  grid-column: 1;
  grid-row: 1 / span 4;
  margin-right: 3em;
}

#cart .vacationImage img {
  object-fit: cover;
  width: 200px;
  height: 250px;
}

#airbnbs {
  grid-template-columns: 1fr 1fr 1fr;
}

#airbnbs .article .vacationTitle, #airbnbs .article .vacationHouseType, #airbnbs .vacationLocation {
  padding: 0 10px;
}

#airbnbs .vacationImage img {
  object-fit: cover;
  width: 100%;
  height: 150px;
}

#airbnbs .vacationLocation { 
  padding-bottom: 20px; 
}

#airbnbs .vacationTitle, #airbnbs .vacationHouseType { 
  margin-bottom: 10px; 
}

.article {
  border: 1px solid #c9cac9;
}

.vacationLocation {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-template-rows: 1fr 1fr;
}

.authorImage {
  grid-column: 1;
  grid-row: 1 / span 2;
  align-self: center;
}

.authorImage img { 
  width: 30px;
  height: 30px;
  padding: 7px;
  border-radius: 50%;
  margin-right: 10px;
}

.vacationLocation .vacationCity {
  grid-column: 2;
  grid-row: 1;
  align-self: end;
}

.vacationCountry {
  grid-column: 2;
  grid-row: 2;
  font-size: 0.5em;
}

.vacationLocation .vacationCountry {
  grid-column: 2;
  grid-row: 2;
}

.vacationLocation .vacationCity, .vacationLocation .vacationCountry {
  font-size: 0.7em;
}

.article .title {
  font-weight: 500;
  font-size: 1.3em;
}

.article .vacationHouseType {
  font-size: 1em;
}