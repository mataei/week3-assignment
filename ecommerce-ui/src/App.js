import React from 'react';
import './App.css';
import VacationRentals from './VacationRentals';

function App() {
  
	return (
    <div className="App">
	  <VacationRentals />
    </div>
	);
 
}

export default App;
