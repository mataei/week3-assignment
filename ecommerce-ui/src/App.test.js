import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from 'enzyme';
import App from './App';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow renders', () => {
	const wrapper = shallow(<App />);
	expect(wrapper.props('children')).toBeDefined();
});

it('renders', () => {
	const wrapper = render(<App />);
	expect(wrapper).toMatchSnapshot();
});

it('mounts', () => {
	const wrapper = mount(<App />);
	expect(wrapper).toMatchSnapshot();
});

