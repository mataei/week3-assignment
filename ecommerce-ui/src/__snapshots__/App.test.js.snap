// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounts 1`] = `
<App>
  <div
    className="App"
  >
    <VacationRentals>
      <div
        className="App"
      >
        <div
          id="shoppingCart"
        >
          <h1>
            <span>
              Shopping Cart
            </span>
          </h1>
          <span>
            Total Cost: $
            0
          </span>
          <ShoppingCart
            onDeleteRental={[Function]}
            rentals={Array []}
          >
            <div>
              <ul
                className="allShoppings"
                id="cart"
              />
            </div>
          </ShoppingCart>
        </div>
        <div
          id="vacationRentals"
        >
          <h1>
            <span>
              Explore Airbnb
            </span>
          </h1>
          <ul
            className="allVacationRentals"
            id="airbnbs"
          >
            <VacationRental
              index={0}
              key="0"
              onBookRental={[Function]}
              vacationRental={
                Object {
                  "host": Object {
                    "isSuperhost": true,
                    "name": "Robert",
                  },
                  "houseType": "Apartment",
                  "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
                  "location": Object {
                    "city": "New York",
                    "country": "USA",
                  },
                  "payment": Object {
                    "cost": 105,
                    "description": "Free cancellation",
                  },
                  "rating": Object {
                    "reviews": 133,
                    "stars": 3,
                  },
                  "title": "Centrally-located Manhattan studio",
                }
              }
            >
              <div
                className="vacationRental"
              >
                <li>
                  <div
                    className="vacationImage"
                  >
                    <img
                      src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
                    />
                  </div>
                  <h2
                    className="vacationTitle"
                  >
                    Centrally-located Manhattan studio
                  </h2>
                  <p
                    className="vacationHouseType"
                  >
                    House Type: 
                    Apartment
                  </p>
                  <div
                    className="vacationLocation"
                  >
                    <div
                      className="vacationCity"
                    >
                      City: 
                      New York
                    </div>
                    <div
                      className="vacationCountry"
                    >
                      Country: 
                      USA
                    </div>
                  </div>
                  <div
                    className="vacationLocation"
                  >
                    <div
                      className="vacationCity"
                    >
                      Cost: $
                      105
                    </div>
                    <div
                      className="vacationCountry"
                    >
                      Free cancellation
                    </div>
                  </div>
                  <button
                    onClick={[Function]}
                  >
                    Book
                  </button>
                </li>
              </div>
            </VacationRental>
            <VacationRental
              index={1}
              key="1"
              onBookRental={[Function]}
              vacationRental={
                Object {
                  "host": Object {
                    "isSuperhost": true,
                    "name": "Jamie",
                  },
                  "houseType": "Apartment",
                  "image": "https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80",
                  "location": Object {
                    "city": "Denver",
                    "country": "USA",
                  },
                  "payment": Object {
                    "cost": 130,
                    "description": "",
                  },
                  "rating": Object {
                    "reviews": 381,
                    "stars": 4.5,
                  },
                  "title": "Denver apartment outside Union Station",
                }
              }
            >
              <div
                className="vacationRental"
              >
                <li>
                  <div
                    className="vacationImage"
                  >
                    <img
                      src="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
                    />
                  </div>
                  <h2
                    className="vacationTitle"
                  >
                    Denver apartment outside Union Station
                  </h2>
                  <p
                    className="vacationHouseType"
                  >
                    House Type: 
                    Apartment
                  </p>
                  <div
                    className="vacationLocation"
                  >
                    <div
                      className="vacationCity"
                    >
                      City: 
                      Denver
                    </div>
                    <div
                      className="vacationCountry"
                    >
                      Country: 
                      USA
                    </div>
                  </div>
                  <div
                    className="vacationLocation"
                  >
                    <div
                      className="vacationCity"
                    >
                      Cost: $
                      130
                    </div>
                    <div
                      className="vacationCountry"
                    />
                  </div>
                  <button
                    onClick={[Function]}
                  >
                    Book
                  </button>
                </li>
              </div>
            </VacationRental>
            <VacationRental
              index={2}
              key="2"
              onBookRental={[Function]}
              vacationRental={
                Object {
                  "host": Object {
                    "isSuperhost": false,
                    "name": "Olivia",
                  },
                  "houseType": "Entire House",
                  "image": "https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80",
                  "location": Object {
                    "city": "Stockholm",
                    "country": "Sweden",
                  },
                  "payment": Object {
                    "cost": 230,
                    "description": "Free cancellation",
                  },
                  "rating": Object {
                    "reviews": 83,
                    "stars": 5,
                  },
                  "title": "Quaint 3-bedroom home in Gamla stan",
                }
              }
            >
              <div
                className="vacationRental"
              >
                <li>
                  <div
                    className="vacationImage"
                  >
                    <img
                      src="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
                    />
                  </div>
                  <h2
                    className="vacationTitle"
                  >
                    Quaint 3-bedroom home in Gamla stan
                  </h2>
                  <p
                    className="vacationHouseType"
                  >
                    House Type: 
                    Entire House
                  </p>
                  <div
                    className="vacationLocation"
                  >
                    <div
                      className="vacationCity"
                    >
                      City: 
                      Stockholm
                    </div>
                    <div
                      className="vacationCountry"
                    >
                      Country: 
                      Sweden
                    </div>
                  </div>
                  <div
                    className="vacationLocation"
                  >
                    <div
                      className="vacationCity"
                    >
                      Cost: $
                      230
                    </div>
                    <div
                      className="vacationCountry"
                    >
                      Free cancellation
                    </div>
                  </div>
                  <button
                    onClick={[Function]}
                  >
                    Book
                  </button>
                </li>
              </div>
            </VacationRental>
          </ul>
        </div>
      </div>
    </VacationRentals>
  </div>
</App>
`;

exports[`renders 1`] = `
<div
  class="App"
>
  <div
    class="App"
  >
    <div
      id="shoppingCart"
    >
      <h1>
        <span>
          Shopping Cart
        </span>
      </h1>
      <span>
        Total Cost: $0
      </span>
      <div>
        <ul
          class="allShoppings"
          id="cart"
        />
      </div>
    </div>
    <div
      id="vacationRentals"
    >
      <h1>
        <span>
          Explore Airbnb
        </span>
      </h1>
      <ul
        class="allVacationRentals"
        id="airbnbs"
      >
        <div
          class="vacationRental"
        >
          <li>
            <div
              class="vacationImage"
            >
              <img
                src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
              />
            </div>
            <h2
              class="vacationTitle"
            >
              Centrally-located Manhattan studio
            </h2>
            <p
              class="vacationHouseType"
            >
              House Type: Apartment
            </p>
            <div
              class="vacationLocation"
            >
              <div
                class="vacationCity"
              >
                City: New York
              </div>
              <div
                class="vacationCountry"
              >
                Country: USA
              </div>
            </div>
            <div
              class="vacationLocation"
            >
              <div
                class="vacationCity"
              >
                Cost: $105
              </div>
              <div
                class="vacationCountry"
              >
                Free cancellation
              </div>
            </div>
            <button>
              Book
            </button>
          </li>
        </div>
        <div
          class="vacationRental"
        >
          <li>
            <div
              class="vacationImage"
            >
              <img
                src="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
              />
            </div>
            <h2
              class="vacationTitle"
            >
              Denver apartment outside Union Station
            </h2>
            <p
              class="vacationHouseType"
            >
              House Type: Apartment
            </p>
            <div
              class="vacationLocation"
            >
              <div
                class="vacationCity"
              >
                City: Denver
              </div>
              <div
                class="vacationCountry"
              >
                Country: USA
              </div>
            </div>
            <div
              class="vacationLocation"
            >
              <div
                class="vacationCity"
              >
                Cost: $130
              </div>
              <div
                class="vacationCountry"
              />
            </div>
            <button>
              Book
            </button>
          </li>
        </div>
        <div
          class="vacationRental"
        >
          <li>
            <div
              class="vacationImage"
            >
              <img
                src="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
              />
            </div>
            <h2
              class="vacationTitle"
            >
              Quaint 3-bedroom home in Gamla stan
            </h2>
            <p
              class="vacationHouseType"
            >
              House Type: Entire House
            </p>
            <div
              class="vacationLocation"
            >
              <div
                class="vacationCity"
              >
                City: Stockholm
              </div>
              <div
                class="vacationCountry"
              >
                Country: Sweden
              </div>
            </div>
            <div
              class="vacationLocation"
            >
              <div
                class="vacationCity"
              >
                Cost: $230
              </div>
              <div
                class="vacationCountry"
              >
                Free cancellation
              </div>
            </div>
            <button>
              Book
            </button>
          </li>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
